test/controller/company.number.controller.spec.unit.ts

```diff
@@ -0,0 +1,102 @@
+jest.mock("../../src/middleware/authentication.middleware");
+jest.mock("../../src/middleware/session.middleware");
+jest.mock("../../src/services/company.profile.service");
+jest.mock("../../src/services/objections.session.service");
+
+import { Session } from "ch-node-session-handler/lib/session/model/Session";
+import { NextFunction, Request, Response } from "express";
+import request from "supertest";
+import app from "../../src/app";
+import authenticationMiddleware from "../../src/middleware/authentication.middleware";
+import sessionMiddleware from "../../src/middleware/session.middleware";
+import ObjectionCompanyProfile from "../../src/model/objection.company.profile";
+import { COMPANY_NUMBER, OBJECTIONS_COMPANY_NUMBER, OBJECTIONS_CONFIRM_COMPANY } from "../../src/model/page.urls";
+import { getCompanyProfile } from "../../src/services/company.profile.service";
+import { COOKIE_NAME } from "../../src/utils/properties";
+
+import { getValidAccessToken } from "../../src/services/objections.session.service";
+
+const ACCESS_TOKEN = "KGGGUYUYJHHVK1234";
+
+const mockAuthenticationMiddleware = authenticationMiddleware as jest.Mock;
+mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+const mockSessionMiddleware = sessionMiddleware as jest.Mock;
+mockSessionMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => {
+    req.session = {
+        data: {},
+    } as Session;
+    return next();
+});
+
+describe("company number lookup tests", () => {
+
+    const mockCompanyProfile = getCompanyProfile as jest.Mock;
+
+    beforeEach(() => {
+        mockCompanyProfile.mockReset();
+    });
+
+    // TODO - Implement generic error page when company search failure handling is implemented
+    //        based on story requirements
+
+    // it("should return the error page when company search fails", async () => {
+    //     mockCompanyProfile.mockImplementation(() => {
+    //         throw {
+    //             message: "Unexpected error",
+    //             status: 500,
+    //         };
+    //     });
+
+    //     const response = await request(app)
+    //         .post(OBJECTIONS_COMPANY_NUMBER)
+    //         .set("Accept", "application/json")
+    //         .set("Referer", "/")
+    //         .set("Cookie", [`${COOKIE_NAME}=123`])
+    //         .send({companyNumber: "00012345"});
+
+    //     expect(response.status).toEqual(500);
+    //     expect(response).not.toBeUndefined();
+    // });
+
+    it("should redirect to the check company details screen when company is found", async () => {
+
+        const mockValidAccessToken = getValidAccessToken as jest.Mock;
+
+        beforeEach(() => {
+            mockValidAccessToken.mockReset();
+        });
+
+        mockValidAccessToken.mockReset();
+        mockValidAccessToken.mockImplementation(() => ACCESS_TOKEN );
+
+        mockCompanyProfile.mockResolvedValueOnce(dummyCompanyProfile);
```

Comment:
mockCompanyProfile.mockResolvedValueOnce(dummyCompanyProfile);  is returning the function rather than executing the function, so in the controller, a function is being added to the session instead of the company profile object.
if you change it to 
mockCompanyProfile.mockResolvedValueOnce(dummyCompanyProfile());
with the extra () - it will execute teh function and return teh company profile.  then you can add an expect(company name).toEqual("blah).  but either take the apostrophe out of the name or in teh expect translate the apostrophe to teh html code for apostrophe.

Reply:
or change getDummyCompanProfile to be just an object instead of a function - see confirm.company.controller test - either way is ok.

Reply:
changed to dummyCompanyProfile